libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/LaneIconProcessor.kt

```diff
@@ -1,237 +1,310 @@
-     * below checks if [activeDirection] contains any string that contains "left" or "right" instead
-     * of checking for string equality.
-     */
-    @DrawableRes
-    fun getDrawableFrom(laneIndicator: LaneIndicator, activeDirection: String?): Int? {
+    fun getLaneIcon(laneIndication: LaneIndicator): LaneIcon {
+        val pair = convertDirectionsToLaneName(laneIndication)
+        val laneName = pair.first
+        val shouldFlip = pair.second
         return when {
-            laneIndicator.directions.size > 2 &&
-                laneIndicator.directions.contains(LEFT) &&
-                laneIndicator.directions.contains(STRAIGHT) &&
-                laneIndicator.directions.contains(RIGHT) -> {
-                activeDirection?.let {
-                    when {
-                        it.contains(LEFT, ignoreCase = true) -> {
-                            laneIcon[LaneTurns.LANE_LEFT_STRAIGHT_RIGHT_LEFT_ONLY]
-                        }
-                        it.contains(RIGHT, ignoreCase = true) -> {
-                            laneIcon[LaneTurns.LANE_LEFT_STRAIGHT_RIGHT_RIGHT_ONLY]
-                        }
-                        else -> {
-                            laneIcon[LaneTurns.LANE_LEFT_STRAIGHT_RIGHT_STRAIGHT_ONLY]
-                        }
-                    }
-                }
+            availableLaneNames.contains(laneName) -> {
+                LaneIcon(availableLaneNames[laneName]!!, shouldFlip)
             }
-            laneIndicator.directions.size > 2 &&
-                laneIndicator.directions.contains(SLIGHT_LEFT) &&
-                laneIndicator.directions.contains(STRAIGHT) &&
-                laneIndicator.directions.contains(SLIGHT_RIGHT) -> {
-                activeDirection?.let {
-                    when {
-                        it.contains(LEFT, ignoreCase = true) -> {
-                            laneIcon[
-                                LaneTurns.LANE_SLIGHT_LEFT_STRAIGHT_SLIGHT_RIGHT_SLIGHT_LEFT_ONLY
-                            ]
-                        }
-                        it.contains(RIGHT, ignoreCase = true) -> {
-                            laneIcon[
-                                LaneTurns.LANE_SLIGHT_LEFT_STRAIGHT_SLIGHT_RIGHT_SLIGHT_RIGHT_ONLY
-                            ]
-                        }
-                        else -> {
-                            laneIcon[LaneTurns.LANE_SLIGHT_LEFT_STRAIGHT_SLIGHT_RIGHT_STRAIGHT_ONLY]
-                        }
-                    }
-                }
+            laneIndication.activeDirection != null -> {
+                // Draw "lane_laneIndication.activeDirection_using_laneIndication.activeDirection"
+                val active = laneIndication.activeDirection.replace(RIGHT, TURN).replace(LEFT, TURN)
+                val name = "$LANE $active $USING $active"
+                LaneIcon(availableLaneNames[name]!!, shouldFlip)
```

Comment:
If there was a new lane indication added to the Directions API this code would start crashing for all clients. This is unlikely to happen in this particular scenario (it would be breaking change for the API), but let's take advantage of Kotlin and write a little bit more defensive code with fallbacks here.

Reply:
> The value of activeDirection can never change and there is no possiblity that it can hold any other values than the one stated above.

That's right, this is our trust in the stability of the Directions API. That's what I'm saying wrt defensive programming - trust in documentation and stability of the API is one thing, but since we don't have a proper contract we might as well be on the safe side and verify the availability of the icon anyway since the language does give us this capability at a low cost.

Reply:
What do we do if the name does not exist? In other words, if `active_direction` has a different value than the one stated above? 
```
laneIndication.activeDirection != null -> {
  // Draw "lane laneIndication.activeDirection using laneIndication.activeDirection"
  val active = laneIndication.activeDirection.replace(RIGHT, TURN).replace(LEFT, TURN)
  val name = "$LANE $active $USING $active"
  if (availableLaneNames.contains(name)) {
    LaneIcon(availableLaneNames[name]!!, shouldFlip)
  } else {
  // What do we return here?
  }
}
```

I honestly feel this would be a serious problem on the directions side if they assign a different value to the `active_direction` other than the ones stated above. Because, there cannot be any other possible lane you can take. However, both the platforms also need to align here. /cc @1ec5 for his inputs and how we want to deal with it.

Reply:
[The `indications` documentation](https://docs.mapbox.com/api/navigation/directions/#lane-object) doesn’t reserve the right to emit an arbitrary value in the future as [some other enumeration-like fields](https://docs.mapbox.com/api/navigation/directions/#maneuver-types) do. But let’s suppose the Directions API has a bug where it fails to filter out OpenStreetMap’s `merge_to_left` value and emits `merge to left` as one of the indications. Depending on this codebase’s overall philosophy about failing early versus failing gracefully, it would be reasonable to either assert or fall back to the same fallback being discussed in https://github.com/mapbox/mapbox-navigation-android/pull/4854#discussion_r713131549. I’m leaning towards the latter for iOS.

Reply:
> it would be reasonable to either assert or fall back to the same fallback being discussed in #4854 (comment). I’m leaning towards the latter for iOS.

Good to know that one of the possible values can be `none` as well. I just want to say that whatever we end up choosing as the fallback, let's keep the behavior same on both the platforms. Either we both show `straight` or a `subset of the indication`.

Reply:
I understand that we're aligned we'd rather fall back to something than throw an exception :+1: